File: samples/JavaScript/Json Store/Social/Tag Document.js
session.setAsync(false);

var store = session.getDatabase("playground").getStore("tempsocial");
var s = "";

store.deleteAllDocuments()
// Create a few documents
function createDoc(unid,val,tags) {
  var doc = store.loadDocument(unid,darwino.jstore.Store.DOCUMENT_CREATE);
  if(doc.isNewDocument()) {
	doc.setJson(
  		{_tags: tags, val:val }
	);
    doc.save()
  }
  return doc;
}
// Create new documents and tag them
var doc1 = createDoc("TAG-1","DOC#1",['MA','CA','TX']);
var doc2 = createDoc("TAG-2","DOC#2",['MA','NH']);
var doc3 = createDoc("TAG-3","DOC#3",['FL','VT']);

// Search by Tags
s += "Searching for MA\n";
var c = store.openCursor().tags("MA").range(0,10);
c.find(function(e) {
  s += "  "+e.getString("val")+", tags:"+e.get("_tags")+"\n";
});  
s += "\n";

s += "Searching for MA AND NH\n";
var c = store.openCursor().tags("MA","NH").range(0,10);
c.find(function(e) {
  s += "  "+e.getString("val")+", tags:"+e.get("_tags")+"\n";
});  
s += "\n";

s += "Searching for MA OR FL\n";
var c = store.openCursor().tags("MA","FL").options(darwino.jstore.Cursor.TAGS_OR).range(0,10);
c.find(function(e) {
  s += "  "+e.getString("val")+", tags:"+e.get("_tags")+"\n";
});  
s += "\n";

s += "Searching for VT\n";
var c = store.openCursor().tags("VT").range(0,10);
c.find(function(e) {
  s += "  "+e.getString("val")+", tags:"+e.get("_tags")+"\n";
});  

// Clean-up...
doc1.deleteDocument(); doc2.deleteDocument(); doc3.deleteDocument();

darwino.Utils.setText("content","{0}",s);

File: samples/Java/Json Store/Social/Tag Document.java
final Store store = session.getDatabase("playground").getStore("tempsocial");
String s = ""; final StringBuilder b = new StringBuilder();

store.deleteAllDocuments();

// Create a few documents
Document doc1 = store.loadDocument("TAG-1",Store.DOCUMENT_CREATE).setJsonString("{val: 'DOC#1', _tags: ['MA','CA','TX']}").save();
Document doc2 = store.loadDocument("TAG-2",Store.DOCUMENT_CREATE).setJsonString("{val: 'DOC#2', _tags: ['MA','NH']}").save();
Document doc3 = store.loadDocument("TAG-3",Store.DOCUMENT_CREATE).setJsonString("{val: 'DOC#3', _tags: ['FL','VT']}").save();

// Search by Tags
s += "Searching for MA\n";
Cursor c = store.openCursor().tags("MA").range(0,10);
b.setLength(0);
c.find(new CursorHandler() {
  public boolean handle(CursorEntry e) throws JsonException {  
  	b.append("  "+e.getString("val")+", tags:"+e.get("_tags")+"\n");
    return true;
  }
});  
s += b.toString()+"\n";

s += "Searching for MA AND NH\n";
c = store.openCursor().tags("MA","NH").range(0,10);
b.setLength(0);
c.find(new CursorHandler() {
  public boolean handle(CursorEntry e) throws JsonException {  
  	b.append("  "+e.getString("val")+", tags:"+e.get("_tags")+"\n");
    return true;
  }
});  
s += b.toString()+"\n";

s += "Searching for MA OR FL\n";
c = store.openCursor().tags("MA","FL").options(Cursor.TAGS_OR).range(0,10);
b.setLength(0);
c.find(new CursorHandler() {
  public boolean handle(CursorEntry e) throws JsonException {  
  	b.append("  "+e.getString("val")+", tags:"+e.get("_tags")+"\n");
    return true;
  }
});  
s += b.toString()+"\n";

s += "Searching for VT\n";
c = store.openCursor().tags("VT").range(0,10);
b.setLength(0);
c.find(new CursorHandler() {
  public boolean handle(CursorEntry e) throws JsonException {  
  	b.append("  "+e.getString("val")+", tags:"+e.get("_tags")+"\n");
    return true;
  }
});  
s += b.toString()+"\n";

// Clean-up...
doc1.deleteDocument(); doc2.deleteDocument(); doc3.deleteDocument();

_formatText("{0}",s);